lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,365 @@
+            - Name of resource group.
+        required: true
+    name:
+        description:
+            - Name of the image.
+        required: true
+    source:
+        description:
+            - OS disk source from the same region, including a virtual machine id or name,
+              OS disk blob uri, managed OS disk id or name, or OS snapshot id or name.
+        required: true
+    data_disk_sources:
+        description:
+            - List of data disk sources, including unmanaged blob uri, managed disk id or name, or snapshot id or name.
+    location:
+        description:
+            - Location of the image.
+    os_type:
+        description: The OS type of image.
+        choices:
+            - Windows
+            - Linux
+    state:
+        description:
+            - Assert the state of the image. Use 'present' to create or update a image and 'absent' to delete an image.
+        default: present
+        choices:
+            - absent
+            - present
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Yuwei Zhou (@yuwzho)"
+
+'''
+
+EXAMPLES = '''
+- name: Create an image from a virtual machine
+  azure_rm_image:
+    resource_group: Testing
+    name: foobar
+    source: testvm001
+
+- name: Create an image from os disk
+  azure_rm_image:
+    resource_group: Testing
+    name: foobar
+    source: /subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing/providers/Microsoft.Compute/disks/disk001
+    data_disk_sources:
+        - datadisk001
+        - datadisk002
+    os_type: Linux
+
+- name: Delete an image
+  azure_rm_image:
+    state: absent
+    resource_group: Testing
+    name: foobar
+    source: testvm001
+'''
+
+RETURN = '''
+state:
+    description: Current state of the image.
+    returned: success
+    type: complex
+    contains:
+        id:
+          description: Image resource path.
+          type: str
+          example: "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing/providers/Microsoft.Compute/images/foobar"
+        name:
+          description: Image name.
+          type: str
+          example: "foobar"
+        location:
+          description: Location of the image.
+          type: str
+          example: "eastus"
+        resourceGroup:
+          description: Resource group of the image.
+          type: str
+          example: "Testing"
+        osDisk:
+          description: OS disk id if the image is created from disks.
+          type: Dict
+          example: { managedDisk: "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing/providers/Microsoft.Compute/disks/testvm001" }
+        dataDisks:
+          description: Data disk id if the image is created from disks.
+          type: list
+          example: [{ managedDisk: "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing/providers/Microsoft.Compute/disks/testvm001-datadisk-0" }]
+        sourceVirtualMachine:
```

Comment:
Already know this, don't return.

Reply:
I think this value should be return. If we assume these all things user already knows, we can return the image id only. But it is so weird

Reply:
I don't think it is weird, there are plenty of modules where they don't return values at all. `os_disk`, `data_disks` and after thinking about it more this one is fine to return as it returns the full path of the VM and not just the name. The only change I would make is to only return it if the user set the vm name as the source.